### ISSUE TITLE
separableApprox/chol is broken

### ISSUE THREAD

**Issue by ajt60gaibb** [chebfun/chebfun#1780]
This used to work:
```
A = chebfun2(@(x,y) cos(10*x.*y) + y + x.^2 + sin(10*x.*y));
R = chol( A' * A );
norm( A'*A - R'*R ) 
ans =
   0.002494201668744
```

It is because the `CDR` factorization is not symmetric for `A'*A`: 
```
A = chebfun2(@(x,y) cos(10*x.*y) + y + x.^2 + sin(10*x.*y));
[C, D, R] = cdr( A'*A );
norm( C - R ) 
ans =
   0.046157076339960
```

**Comment by aaustin141** [comment]
@ajt60gaibb:  When you say this "used to work", do you have an idea of when and how well?  The last release on which the code doesn't crash with a "not symmetric" error is v5.1.0 (with commit adcbe30c being the commit released in v5.2.0 that broke things), but for `norm(A'*A - R'*R)`, I get a number on the order of `1e-4`, which is not great.

**Comment by ajt60gaibb** [comment]
This is now fixed in #1779.  It required a small change to `separableApprox/mtimes`, so that the command preserves symmetry if it exists. 

I now get: 

```
A = chebfun2(@(x,y) cos(10*x.*y) + y + x.^2 + sin(10*x.*y));
[C, D, R] = cdr( A'*A );
norm( C - R ) 
ans =
     7.634767857879577e-08
```
which is going to be approximately the error in Cholesky factor of `A'*A`.   

However, we are still able to calculate an accurate Cholesky factorization, as we are accustomed to from the discrete setting: 
``` 
A = chebfun2(@(x,y) cos(10*x.*y) + y + x.^2 + sin(10*x.*y));
R = chol( A' * A );
norm( A'*A - R'*R )
ans =
     0
```
